streamer/pulp/streamer/server.py

```diff
@@ -129,7 +159,7 @@ def _handle_get(self, request):
                 for entry in q_set.all():
                     logger.info('Trying URL: {url}'.format(url=entry.url))
                     try:
-                        last_report = self._download(entry, responder)
+                        last_report = self._download(request, entry, responder)
```

Comment:
Agreed. This is not intuitive.  My understanding is based on the original streamer code.  It seems like only the twisted request is passed to the streamer.  The twisted `Request` seems to bidirectional.  Even looking at the `Responder` it effectively writes the bits using `Request.write()`.  I'll look at twisted docs to make sure.  But, this is how the 2.10 streamer did things and it seems to work.